src/Controller/EnMarche/CitizenActionManagerController.php

```diff
@@ -82,4 +105,156 @@ public function editAction(Request $request, CitizenProject $project, CitizenAct
+
+        if (0 == $registrations->count()) {
+            $this->addFlash('error', $this->get('translator')->trans('citizen_action.export.none'));
+
+            return $this->redirectToRoute('app_citizen_action_list_participants', [
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        $command = new CitizenActionContactParticipantsCommand($registrations->toArray(), $this->getUser());
+
+        $form = $this->createForm(ContactMembersType::class, $command, ['csrf_token_id' => 'citizen_action.contact_participants'])
+            ->add('submit', SubmitType::class)
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(CitizenActionContactParticipantsCommandHandler::class)->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_action.contact.success'));
+
+            return $this->redirectToRoute('app_citizen_action_list_participants', [
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        $uuids = array_map(function (EventRegistration $registration) {
+            return $registration->getUuid()->toString();
+        }, $registrations->toArray());
+
+        return $this->render('citizen_action_manager/contact_participants.html.twig', [
+            'citizen_action' => $citizenAction,
+            'citizen_project' => $project,
+            'contacts' => $uuids,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{slug}/participants/imprimer", name="app_citizen_action_print_participants")
+     * @Method("POST")
+     * @Security("is_granted('EDIT_CITIZEN_ACTION', project)")
+     */
+    public function printParticipantsAction(Request $request, CitizenAction $citizenAction, CitizenProject $project, CitizenActionManager $citizenActionManager): Response
+    {
+        $registrations = $this->getRegistrations($request, $citizenAction, self::ACTION_PRINT);
+
+        if (0 == $registrations->count()) {
+            return $this->redirectToRoute('app_citizen_action_list_participants', [
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        return $this->getPdfResponse(
+            'citizen_action_manager/print_participants.html.twig',
+            [
+                'participants' => $citizenActionManager->populateRegistrationWithAdherentsInformations($registrations),
+            ],
+            'Liste des participants.pdf'
+        );
+    }
+
+    private function getRegistrations(Request $request, CitizenAction $citizenAction, string $action): EventRegistrationCollection
+    {
+        if (!in_array($action, self::ACTIONS)) {
+            throw new \InvalidArgumentException("Action '$action' is not allowed.");
+        }
+
+        if (!$this->isCsrfTokenValid(sprintf('citizen_action.%s_participants', $action), $request->request->get('token'))) {
```

Comment:
how you creat the `citizen_action.export_participants` token id?

Reply:
In the template.